Enabling PWM support for STM32 L152RE Platform

sarthsmart opened the pull request:
Enabled PWM support for STM32 L152RE target platform.

Verified PWM signals on timer3 using test application `tests/drivers/pwm`

sarthsmart committed:
boards: arm: enable PWM support for nucleo_l152re in device tree

Enabling PWM support for STM32 nucleo_l152re in device tree.

Signed-off-by: Sidhdharth Yadav <sidhdharth.yadav@hcl.com>

gmarull commented on dts/arm/st/l1/stm32l1.dtsi:
```diff
@@ -32,6 +32,30 @@
+			compatible = "st,stm32-rcc";
+			#clock-cells = <2>;
+			reg = <0x40023800 0x400>;
+		};
```
rcc/flash changes should go into its own commit. is this change made because others define flash/rcc at the top?

sarthsmart replied:
@gmarull Yes. It is moved from the bottom to top.

FRASTM replied:
yes, this is disturbing to see a change on this part of the dts, could this be avoided ?

sarthsmart replied:
@FRASTM This specific change was made to add nodes after rcc/flash, looks similar to other platform files.

gmarull replied:
@sarthsmart can you move these changes to a separate commit?

FRASTM replied:
+1

sarthsmart replied:
@FRASTM @erwango @gmarull Separate commit for changing the sequence of rcc/flash has been submitted.

FRASTM replied:
thanks, this PR could now be closed

FRASTM commented:
The compilation error in the drivers/pwm/pwm_stm32.c :
It seems the stm32L1 soc series does not support the break function on any of its timer, so should the pwm_stm32.c include a flag on **CONFIG_SOC_SERIES_STM32L1X** in the pwm_stm32_init function:
```
#ifndef CONFIG_SOC_SERIES_STM32L0X
	/* enable outputs and counter */
	if (IS_TIM_BREAK_INSTANCE(cfg->timer)) {
		LL_TIM_EnableAllOutputs(cfg->timer);
	}
#endif
```

sarthsmart committed:
dts: arm: stm32: Move rcc/flash to top for aligning code properly

Sequence of code changed from bottom to top in dtsi file.

Signed-off-by: Sidhdharth Yadav <sidhdharth.yadav@hcl.com>

sarthsmart committed:
dts: arm: stm32: Enable pwm support for stm32l1 in dtsi

Enabling pwm on timer3 for stm32l1 series in dtsi.

Adding other timer nodes for pwm capability.

Signed-off-by: Sidhdharth Yadav <sidhdharth.yadav@hcl.com>

sarthsmart committed:
drivers: pwm: Adding a flag to allow build for stm32l1 series

Include a flag to allow build for stm32l1 series

Signed-off-by: Sidhdharth Yadav <sidhdharth.yadav@hcl.com>
